middleware/allRoutes/handler.go

```diff
@@ -64,8 +72,17 @@ func Handler(routesHandler map[string]http.Handler) func(h http.Handler) http.Ha
 				return
 			}
 
-			b, err := ioutil.ReadAll(res.Body)
+			// Use a limited reader so we dont oom the router checking for content-type
+			limitReader := io.LimitReader(res.Body, int64(config.ContentTypeByteLimit+1))
```

Comment:
unsure of the answer to this... if this was an inbound http request, and we didn't read the entire body, we wouldn't be able to reuse the connection

not sure if that's the same for a http response to an outbound http request - I'd guess it is... so we might want to defer a `io.Copy(ioutil.Discard, res.Body)`?

Reply:
added a `defer io.Copy(ioutil.Discard, res.Body)` on the next line.